lib/libzfs/libzfs_changelist.c

```diff
@@ -541,15 +541,14 @@ change_one(zfs_handle_t *zhp, void *data)
 			uu_avl_insert(clp->cl_tree, cn, idx);
 		} else {
 			free(cn);
-			zfs_close(zhp);
 		}
 
 		if (!clp->cl_alldependents)
 			return (zfs_iter_children(zhp, change_one, data));
```

Comment:
Who closes the handle after we return here?

Reply:
Good point, I kinda assumed that it was done by `zfs_iter_children`, but looking at the code it is evidently not the case. Ideally we would not close the handle in the callback, but rather in the function that created the handle (`zfs_iter_filesystems` and `zfs_iter_snapshots` in this case), but that would require changes in quite a bunch of places.

I will post an update shortly.